tt_metal/impl/device/device_mesh.cpp

```diff
@@ -49,13 +47,17 @@ DeviceMesh::DeviceMesh(const DeviceGrid& device_grid, const DeviceIds &device_id
         }
         managed_devices = tt::tt_metal::detail::CreateDevices(galaxy_device_ids, num_command_queues, l1_small_size, trace_region_size, dispatch_core_type);
         for (int i = 0; i < num_requested_devices; i++) {
-            mesh_devices.emplace_back(device_ids[i], managed_devices.at(galaxy_device_ids[i]));
+            mesh_devices.emplace_back(i, managed_devices.at(galaxy_device_ids[i]));
         }
         this->view = std::make_unique<tt::tt_metal::DeviceMeshView>(*this);
     } else {
+        DeviceIds device_ids;
```

Comment:
do we still need this if/else galaxy switch?

otherwise `this->view = std::make_unique<tt::tt_metal::DeviceMeshView>` needs to go into the switch as well.

Reply:
ah, this is a TODO since we couldn't get this change into this PR?